lib/network/shard-server.js

```diff
@@ -397,7 +401,17 @@ ShardServer.prototype.routeRetrieval = function(req, res) {
 
       function _handleTransferFinish() {
         self.activeTransfers--;
-        self._log.debug('Shard download completed hash %s', item.hash);
+
+        if (req.headers['user-agent']) {
+          self._log.info('Shard download completed hash %s size %s',
+            item.hash,
+            contract.get('data_size'));
+        } else {
```

Comment:
This check for mirror vs. download seems a bit fragile, as it's dependent on user-agent. So we should be aware of this later.